modules/core/include/opencv2/core/cuda.hpp

```diff
@@ -478,29 +478,29 @@ and the corresponding memory is automatically returned to the pool for later usa
     }
 @endcode
  */
-class CV_EXPORTS BufferPool
+class CV_EXPORTS_W BufferPool
 {
 public:
 
     //! Gets the BufferPool for the given stream.
-    explicit BufferPool(Stream& stream);
+    CV_WRAP explicit BufferPool(Stream& stream);
```

Comment:
Generated Python code:
```
    PyObject* pyobj_stream = NULL;
    Stream stream;

    const char* keywords[] = { "stream", NULL };
    if( PyArg_ParseTupleAndKeywords(args, kw, "O:BufferPool", (char**)keywords, &pyobj_stream) &&
        pyopencv_to(pyobj_stream, stream, ArgInfo("stream", 0)) )
    {
        new (&(self->v)) Ptr<cv::cuda::BufferPool>(); // init Ptr with placement new
        if(self) ERRWRAP2(self->v.reset(new cv::cuda::BufferPool(stream)));
        return 0;
    }
```

> Stream stream;

Constructor expects reference, not an another copy of object.

Reply:
Dropped wrapping it for the time being